runtime/common/RuntimeMLIR.cpp

```diff
@@ -201,4 +201,39 @@ void registerToIQMJsonTranslation() {
         return cudaq::translateToIQMJson(op, output);
       });
 }
+
+ExecutionEngine *createQIRJITEngine(ModuleOp &moduleOp) {
+  ExecutionEngineOptions opts;
+  opts.transformer = [](llvm::Module *m) { return llvm::ErrorSuccess(); };
+  opts.jitCodeGenOptLevel = llvm::CodeGenOpt::None;
+  opts.llvmModuleBuilder =
+      [&](Operation *module,
+          llvm::LLVMContext &llvmContext) -> std::unique_ptr<llvm::Module> {
+    llvmContext.setOpaquePointers(false);
+
+    auto *context = module->getContext();
+    PassManager pm(context);
+    std::string errMsg;
+    llvm::raw_string_ostream errOs(errMsg);
+    std::string qirBasePipelineConfig =
+        "func.func(quake-add-deallocs),quake-to-qir";
+    if (failed(parsePassPipeline(qirBasePipelineConfig, pm, errOs)))
+      throw std::runtime_error("[createQIRJITEngine] Lowering to QIR failed.");
```

Comment:
If we leave this here, we need report this error slightly different. This error is not something in the control of the user as it is failing to parse a pipeline that is hardcoded in this function. It needs to be some sort of "runtime internal error".

Alternatively, we could add those passes directly to the pass manager:
```
pm.addNestedPass<func::FuncOp>(cudaq::opt::createQuakeAddDeallocs());
pm.addPass(cudaq::opt::createConvertToQIRPass());
```

Reply:
I like that approach better. Will update.